corehq/toggles/shortcuts.py

```diff
@@ -22,26 +22,45 @@ def set_toggle(slug, item, enabled, namespace=None):
     """
     Sets a toggle value explicitly. Should only save anything if the value needed to be changed.
     """
-    if toggle_enabled(slug, item, namespace=namespace) != enabled:
-        ns_item = namespaced_item(item, namespace)
-        try:
-            toggle_doc = Toggle.get(slug)
-        except ResourceNotFound:
-            toggle_doc = Toggle(slug=slug, enabled_users=[])
-        if enabled:
-            toggle_doc.add(ns_item)
-        else:
-            toggle_doc.remove(ns_item)
-        from corehq.feature_previews import all_previews
-        from corehq.toggles import all_toggles, NAMESPACE_DOMAIN
-        static_toggles_by_slug = {t.slug: t for t in all_toggles() + all_previews()}
-        if namespace == NAMESPACE_DOMAIN and slug in static_toggles_by_slug:
-            static_toggle = static_toggles_by_slug[slug]
-            if static_toggle.save_fn:
-                static_toggle.save_fn(item, enabled)
+    if _set_toggle_without_clear_cache(slug, item, enabled, namespace=namespace) != enabled:
```

Comment:
Thank you for catching that! that could have been bad [7da981e](https://github.com/dimagi/commcare-hq/pull/35332/commits/7da981ef5fd820ee08da75d341c661b338dfc885)

Reply:
oh wait you're right. The equality check for `enabled` was originally for checking whether to actually change the toggle. I got it mixed up since that was moved somewhere else. I was wondering why tests didn't fail.